int32 n
---
bool success